### ISSUE TITLE
Icon missing

### ISSUE THREAD

**Issue by ctron** [flathub/com.spotify.Client#179]
Since the most recent update, the application icon is gone, and replaced by the generic Xorg logo.

**Comment by agurenko** [comment]
Yeah, you can see briefly upon opening and closing the app that it uses the correct icon, but it stays generic during the use.

**Comment by bsmith7581** [comment]
The latest version of the .deb package has the same issue, so it's not an issue with the flatpak.

**Comment by m93a** [comment]
I'm on elementary OS 6 and have Spotify pinned in my Deck. When I open it, it doesn't a window marked as Spotify like it used to. Instead it opens a new unnamed window with a generic gear icon:
![image](https://user-images.githubusercontent.com/1671665/135754761-4706c43e-5e2a-42e1-9e94-05ee36fc864e.png)

I guess this is related.

**Comment by Erick555** [comment]
https://community.spotify.com/t5/Desktop-Linux/Few-broken-things-since-latest-update/td-p/5264045

**Comment by tomaswarynyca** [comment]
Works on ElementaryOS 6 @m93a 
https://community.spotify.com/t5/Desktop-Linux/Few-broken-things-since-latest-update/m-p/5270918/highlight/true#M20766

**Comment by lil5** [comment]
https://community.spotify.com/t5/Desktop-Linux/Few-broken-things-since-latest-update/m-p/5297240/highlight/true#M20834

I saw that someone managed to get it working well on snap is that correct?
If so it might not be an upstream issue after all.

**Comment by Erick555** [comment]
The original report in that thread came from ArchLinux user who didn't use flatpak so it's might be not flatpak issue either. Linux users community is so diverse that it's very hard to make apple-apple comparisons.

Maybe someone can try [workaround](https://aur.archlinux.org/packages/spotify/#comment-836034):

> I found a workaround for the X11 icon in the task switcher issue which works for me on KDE. Click "Full screen" at the bottom-right corner, and optionally click it again to return to the normal mode. The icon becomes the normal Spotify icon.

**Comment by ctron** [comment]
I just tried this workaround, and it seems to restore the icon. However:

* I noticed that the icon shows up eventually. Setting the current title in the description. However, it gets stuck too. Meaning, it doesn't update the title any longer. The workaround doesn't refresh the title.
* I also notice, when closing spotify, that a second window appears briefly. Fully black. Is closes quickly and spotify exits.

I am using Fedora 35 with KDE.
